packages/flutter/lib/src/widgets/scrollbar.dart

```diff
@@ -1211,18 +1225,37 @@ class RawScrollbarState<T extends RawScrollbar> extends State<T> with TickerProv
     );
   }
 
+  bool _handleScrollContentMetricsNotification(ScrollContentMetricsNotification notification) {
+    if (!widget.notificationPredicate(ScrollUpdateNotification(metrics: notification.metrics, context: notification.context)))
```

Comment:
The type-cast here a little hack.

Reply:
I think it is ok though. I thought maybe we could  explore changing the typing of a ScrollNotificationPredicate to take a LayoutChangedNotification, both use cases are extensions of that base class, but it appears it would be pretty breaking.

Reply:
Or, could `ScrollContentMetricsNotification` extend `ScrollNotification`? It is part of the scrolling family, so maybe we should have it inherit instead. I think that may be cleaner, and it'll take care of this cast too.

Reply:
I realize that might be breaking as well, since anyone who listens to a ScrollNotification would need to account for this new kind of ScrollNotification. May still be worth checking out.

Reply:
```dart
I realize that might be breaking as well, since anyone who listens to a ScrollNotification would need to account for this new kind of ScrollNotification. May still be worth checking out.
```
This actually breaking a lot of unit tests.
I prefer to keep it as is until we find a better solution if you agree.